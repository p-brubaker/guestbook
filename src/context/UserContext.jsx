import { createContext, useState, useContext, useEffect } from 'react';

const UserContext = createContext();

function UserProvider({ children }) {
  const [user, setUser] = useState(null);

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
}

function useUser() {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser hook must be called within a UseProvider');
  }

  return context;
}

export { useUser, UserProvider };
