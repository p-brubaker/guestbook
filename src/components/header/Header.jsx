import './Header.css';
import React from 'react';
import { useUser } from '../../context/UserContext';

export default function Header() {
  const { user } = useUser();

  return user ? (
    <div className="header">
      <p>Thanks for signing in {user}!</p>
    </div>
  ) : (
    <div className="header">
      <p>Sign in Friend!</p>
    </div>
  );
}
