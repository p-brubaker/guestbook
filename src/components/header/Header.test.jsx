import { render, screen } from '@testing-library/react';
import Header from './Header';
import React from 'react';
import { UserProvider } from '../../context/UserContext';

it('should render the header component', async () => {
  const { container } = render(
    <UserProvider>
      <Header />
    </UserProvider>
  );

  screen.getByText('Sign in Friend!');

  expect(container).toMatchSnapshot();
});
