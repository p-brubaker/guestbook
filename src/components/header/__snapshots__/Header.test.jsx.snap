// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the header component 1`] = `
<div>
  <div
    class="header"
  >
    <p>
      Sign in Friend!
    </p>
  </div>
</div>
`;
